nvflare/private/fed/client/client_runner.py

```diff
@@ -404,6 +407,17 @@ def _try_run(self):
 
                 time.sleep(task_fetch_interval)
 
+    def send_job_heartbeat(self, interval=30.0):
+        wait_times = int(interval / 2)
+        while not self.asked_to_stop:
+            with self.engine.new_context() as fl_ctx:
+                self.engine.send_job_heartbeat(fl_ctx)
+
+                for i in range(wait_times):
```

Comment:
An event might work better here. An example can be found here: https://github.com/NVIDIA/NVFlare/blob/ab0c7296cbe57568ec0f7c1f07c321eeb44f9787/nvflare/fuel/f3/sfm/heartbeat_monitor.py#L47-L54

Reply:
Keep the time interval checking for now.